### PR TITLE
Fix failures in iceberg cloud tests

### PR DESCRIPTION
**Pull request by lukasz-stec**
## Description
Fixes ci failures introduced by https://github.com/trinodb/trino/pull/24469

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
## Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by raunaqmorarka** [comment]
/test-with-secrets sha=9e18af13e7f8d91fa73d491e162ebca2505cf2b4

**Comment by lukasz-stec** [comment]
added missing @Test annotation

**Comment by hashhar** [comment]
/test-with-secrets sha=a1b9d9067fd24eb591015bfcb34874f85aed7663

**Comment by hashhar** [comment]
Run with secrets is green.

**Review by hashhar — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestIcebergSnowflakeCatalogConnectorSmokeTest.java [hunk 1]
```diff
@@ -684,6 +684,14 @@ public void testSnowflakeNativeTable()
                 .hasRootCauseMessage("SQL compilation error:\ninvalid parameter 'table ? is not a Snowflake iceberg table'");
     }
 
+    @Test
+    @Override
+    public void testIcebergTablesFunction()
+    {
+        assertThatThrownBy(super::testIcebergTablesFunction)
+                .hasMessageContaining("schemaPath is not supported for Iceberg snowflake catalog");
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestIcebergSnowflakeCatalogConnectorSmokeTest.java:692]: Snowflake catalog supports `iceberg_tables` function, right? Skipping the entire test doesn't make sense to me. We could run this function without additional schema creation.

Comment (lukasz-stec) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/snowflake/TestIcebergSnowflakeCatalogConnectorSmokeTest.java:692]: @ebyhr Right, I did it quickly so the CI would be green + it follows a pattern in this test class.
I submitted a new [PR](https://github.com/trinodb/trino/pull/24593) that actually tests this for the snowflake catalog. PTAL

**Pull request merged**